public/app/features/scenes/dashboard/PanelTimeRange.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import { dateMath, getTimeZone, rangeUtil, TimeRange } from '@grafana/data';
+import {
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneTimeRangeLike,
+  SceneTimeRangeState,
+} from '@grafana/scenes';
+import { TimeZone } from '@grafana/schema';
+import { TimeOverrideResult } from 'app/features/dashboard/utils/panel';
+
+interface PanelTimeRangeState extends SceneTimeRangeState {
+  timeFrom?: string;
+  timeShift?: string;
+}
+
+export class PanelTimeRange extends SceneObjectBase<PanelTimeRangeState> implements SceneTimeRangeLike {
```

Comment:
Since this is to support current behavior, I think it's fine to have it in core.

Some time ago I've implemented this https://github.com/grafana/grafana/pull/54985/files#diff-0dd2633669aa977794bb87162fa51272993c3aa9da441c0cc9d6b1fe13733eaaR13 that was supposed to allow time shifts to enable i.ee WoW/YoY and similar comparisons - I can imagine such primitive living in @scenes package at some point.